core/src/main/scala/org/apache/spark/sql/delta/commands/ShowCreateTableCommand.scala

```diff
@@ -0,0 +1,156 @@
+import org.apache.spark.sql.connector.catalog.{CatalogV2Util, Table, TableCatalog}
+import org.apache.spark.sql.catalyst.util.escapeSingleQuotedString
+import org.apache.spark.sql.delta.DeltaErrors.cannotShowCreateGeneratedColumnsProperty
+import org.apache.spark.sql.delta.catalog.DeltaTableV2
+import org.apache.spark.sql.delta.sources.DeltaSourceUtils
+import org.apache.spark.sql.execution.command.LeafRunnableCommand
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.unsafe.types.UTF8String
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * A command for describing the details of a table such as the format, name, and size.
+ * Implementation partially reused
+ * from [https://github.com/apache/spark/blob/branch-3.3/
+ * sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala]
+ */
+case class ShowCreateTableCommand(table: Option[TableIdentifier])
+  extends LeafRunnableCommand with DeltaCommand {
+
+  case class ShowCreateTableOutput(ddl: String)
+
+  override val output: Seq[Attribute] = ExpressionEncoder[ShowCreateTableOutput]().
+    schema.toAttributes
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val builder = new mutable.StringBuilder
+    val md = sparkSession.sessionState.catalog.getTableMetadata(table.get)
+    val dt = DeltaTableV2(sparkSession, new Path(md.location), Option(md))
+    showCreateTable(dt, builder)
+    Seq(Row(UTF8String.fromString(builder.toString)))
+  }
+
+  private def showCreateTable(table: Table, builder: mutable.StringBuilder): Unit = {
+    builder ++= s"CREATE TABLE ${table.name()} "
+
+    showTableDataColumns(table, builder)
+    showTableUsing(table, builder)
+
+    val tableOptions = table.properties.asScala
+      .filterKeys(_.startsWith(TableCatalog.OPTION_PREFIX)).map {
+      case (k, v) => k.drop(TableCatalog.OPTION_PREFIX.length) -> v
+    }.toMap
+    showTableOptions(builder, tableOptions)
+    showTablePartitioning(table, builder)
+    showTableComment(table, builder)
+    showTableLocation(table, builder)
+    showTableProperties(table, builder, tableOptions)
+  }
+
+  private def showTableDataColumns(table: Table, builder: mutable.StringBuilder): Unit = {
+    val tableSchema = table.asInstanceOf[DeltaTableV2].deltaLog.snapshot.metadata.schema
+    if (tableSchema.exists(column => column.metadata.contains(
+      DeltaSourceUtils.GENERATION_EXPRESSION_METADATA_KEY))) {
+      throw cannotShowCreateGeneratedColumnsProperty(table.name())
+    }
+    val columns = CharVarcharUtils.getRawSchema(table.schema()).fields.map(_.toDDL)
+    builder ++= concatByMultiLines(columns)
+  }
+
+  private def showTableUsing(table: Table, builder: mutable.StringBuilder): Unit = {
+    Option(table.properties.get(TableCatalog.PROP_PROVIDER))
+      .map("USING " + escapeSingleQuotedString(_) + "\n")
+      .foreach(builder.append)
+  }
```

Comment:
nit: We can hard code DELTA here since we shouldn't run the code with non Delta format